### PR TITLE
Clear web storage gracefully

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #6237 

## Relevant technical choices

This PR updates the way how web storage is currently cleared in Site Kit so that instead of clearing the entire cache, only caches added by Site Kit is removed.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit-base.js
Comment (aaemnnosttv) [inline assets/js/googlesitekit-base.js:24]: This issue has #6250 as a dependency to avoid this, as we do not want to include this here. This whole file is going away so this change is only a problem if this were released before it's dependent issue.

File: assets/js/googlesitekit-main-dashboard.js
Comment (aaemnnosttv) [inline assets/js/googlesitekit-main-dashboard.js:40]: Following my previous comment, this is an example where the function isn't async but thinking again we should probably use this consistently and await it anyways since we're following this call with others. The only time this shouldn't need to be awaited would be as the only function attached to an event handler for example.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @nfmohit – this is almost there, but also we can't merge it yet due to its dependency.

File: assets/js/components/ResetButton.js
Comment (nfmohit) [reply assets/js/components/ResetButton.js:107]: Understood. Addressed, thank you for pointing it out!

File: assets/js/googlesitekit-main-dashboard.js
Comment (nfmohit) [reply assets/js/googlesitekit-main-dashboard.js:40]: Understood. Addressed, thank you for pointing it out!

File: assets/js/googlesitekit-base.js
Comment (nfmohit) [reply assets/js/googlesitekit-base.js:24]: Understood and completely agreed. I'll keep this issue in Execution for now until the dependent is merged, after which I'll go ahead and update the branch to be up to date with `develop`. That way, I don't have to make additional changes, like bringing back the already removed `clearWebStorage`.

Comment (nfmohit) [reply assets/js/googlesitekit-base.js]: This is back in CR for another go, thanks!

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks @nfmohit !

**Pull request merged**
